packages/storybook/stories/components/modal/modal.web-components.stories.ts

```diff
@@ -69,9 +70,27 @@ const TemplateDefault = (args:any) => {
     }
   }
 
+  const locationChangeTrigger = () => {
```

Comment:
did you try instead to listen to the location change (with for eample a MutationObserver) instead of running a interval?
connect this watcher on page load

Reply:
That was a good advice ! I learned something today. Thanks. Code updated accordingly,